webpack.config.js

```diff
@@ -0,0 +1,155 @@
+const webpack = require('webpack');
+const TerserPlugin = require('terser-webpack-plugin');
+const path = require('path');
+const argv = require('yargs').argv;
+const pkg = require('./package');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+
+const IS_PRODUCTION = Boolean(argv.production);
+const VERSION = argv.production ? pkg.version : 'dev';
+const SOURCE_DIR = path.join(__dirname, 'src');
+const DIST_DIR = path.join(__dirname, 'dist');
+const VERSIONED_DIST_DIR = path.join(DIST_DIR, VERSION);
+const SOURCE_DOCS_DIR = path.join(SOURCE_DIR, 'docs');
+const SOURCE_COMPONENTS_DIR = path.join(SOURCE_DIR, 'components');
```

Comment:
I'd create a separate config file with all the paths

Reply:
i found that it's quite hard to reason about paths in our main app, when our configs are split into different files

Reply:
so maybe 
```
const config = {
  ...
}
```

Reply:
config for config seems weird, normal constants would be enough :D

Reply:
My experience is that this `path.join('src', 'docs')` is an overkill and actually makes working with the config way harder. In this case I would just go for strings `'src/docs/'`.